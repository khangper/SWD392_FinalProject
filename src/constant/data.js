export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
