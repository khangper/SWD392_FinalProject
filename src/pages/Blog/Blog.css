/*Insights, ideas, and stories*/
.blog-about,
.blog-blog,
.blog-company,
.blog-careers,
.blog-press {
  position: relative;
  line-height: 20px;
  font-weight: 550;
  color: #424242;
  display: inline-block;
}

.blog-navigation-links,
.blog-navigation-links-2,
.blog-navigation-links-3,
.blog-navigation-links-4 {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
}

.blog-navigation-links-1 {
  border-bottom: 2px solid #ed2a26;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-content: flex-start;
}

.blog-navigation-links,
.blog-navigation-links-1,
.blog-navigation-links-2,
.blog-navigation-links-3,
.blog-navigation-links-4 {
  padding: 6px 20px;
}

.blog-navigation-links-parent {
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  flex-direction: row;
  max-width: 100%;
  font-size: 14px;
}

.blog-heading-2 {
  margin: 0;
  position: relative;
  font-size: 21px;
  line-height: 30px;
  font-weight: 600;
  font-family: inherit;
  color: #333;
  flex-shrink: 0;
}

.blog-heading-2-wrapper {
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-content: flex-end;
  padding: 0 44px 0 47px;
  box-sizing: border-box;
  max-width: 100%;
  text-align: center;
  font-size: 24px;
}

.blog-background {
  align-self: stretch;
  background-image: url("../../assets/background_about.jpg");
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  padding: 30px 20px 55px;
  box-sizing: border-box;
  position: relative;
  gap: 65px;
  min-height: 216px;
  max-width: 100%;
  text-align: left;
  font-size: 14px;
  color: #333;
}
/*Follow*/

.blog-icon-search {
  width: 20px;
  height: 20px;
  color: #686f7a;
}

.blog-button-search {
  padding: 8px 5px 5px 18px;
  border: 0;
  background-color: transparent;
}

.blog-input,
.blog-input-child,
.blog-item {
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
}

.blog-input-child {
  border: 0;
  outline: 0;
  background-color: transparent;
  width: 100%;
  flex-direction: column;
  padding: 10px;
  box-sizing: border-box;
  font-size: 14px;
}

.blog-input,
.blog-item {
  flex-direction: row;
}

.blog-input {
  flex: 1;
  box-shadow: 0 1px 4px 4px rgba(0, 0, 0, 0.02) inset;
  border-radius: 3px;
  background-color: #fff;
  border: 1px solid #e5e5e5;
  gap: 1.4px;
}

.blog-input:hover {
  border: 1px solid #333;
  color: #333;
}

.blog-labels,
.blog-symbol {
  position: relative;
  font-size: 14px;
  line-height: 20px;
  text-align: left;
}

.blog-labels {
  font-family: inherit;
  color: #333;
  display: inline-block;
  min-width: 42px;
}

.blog-symbol {
  height: 20px;
  width: 5.3px;
  color: #a7a8aa;
  display: flex;
  align-items: center;
  flex-shrink: 0;
}
.blog-item {
  align-self: stretch;
  border-bottom: 1px solid #efefef;
  padding: 15px 15px 14px;
}

.blog-filter-content,
.blog-filter-dropdown,
.blog-filter-dropdown-1 {
  align-self: stretch;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
}

.blog-filter-content {
  flex: 1;
  justify-content: space-between;
  gap: 20px;
}

.blog-filter-dropdown,
.blog-filter-dropdown-1 {
  height: 40px;
  justify-content: flex-end;
  padding: 0 22px 0 20px;
  box-sizing: border-box;
  cursor: pointer;
}

.blog-filter-dropdown {
  border-bottom: 1px solid #efefef;
}

.blog-item-1 {
  align-self: stretch;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  border-bottom: 1px solid #efefef;
  padding: 14px 0 0;
  gap: 15px;
}

.blog-follow-twitter,
.blog-icon-twitter,
.blog-follow-facebook,
.blog-icon-facebook {
  position: relative;
}

.blog-icon-twitter,
.blog-icon-facebook {
  height: 17px;
  width: 17px;
  flex-shrink: 0;
  color: white;
}

.blog-follow-twitter,
.blog-follow-facebook {
  font-size: 14px;
  line-height: 17px;
  font-weight: 550;
  color: #fff;
  text-align: center;
  display: inline-block;
  min-width: 42px;
}

.blog-button-1,
.blog-button-2 {
  cursor: pointer;
  border: 0;
  padding: 6.9px 20px 6.1px 24px;
  align-self: stretch;
  border-radius: 3px;
  display: flex;
  flex-direction: row;
  align-items: flex-end;
  justify-content: center;
  gap: 9.8px;
}

.blog-button-1 {
  background-color: #1da1f2;
}

.blog-button-2 {
  background-color: #3b5998;
}

.blog-item-2,
.blog-filter-item {
  align-self: stretch;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
}

.blog-item-2 {
  border-bottom: 1px solid #efefef;
  padding: 20px 20px 19px;
  gap: 10px;
  z-index: 1;
  margin-top: -0.1px;
}

.blog-filter-item {
  padding: 0 0 15px;
}

.blog-learn-more {
  position: relative;
  font-size: 15px;
  line-height: 26px;
  color: #686f7a;
  text-align: center;
  display: inline-block;
  min-width: 71px;
  white-space: nowrap;
}

.blog-learn-more-link {
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-content: flex-start;
  padding: 0 25px 0 24px;
}

.blog-help-center {
  position: relative;
  font-size: 15px;
  line-height: 26px;
  color: #4183c4;
  text-align: center;
  display: inline-block;
  min-width: 121px;
  cursor: pointer;
}

.blog-help-center:hover {
  color: black;
}

.blog-help-links,
.blog-help-links-wrapper {
  display: flex;
  align-items: flex-start;
}

.blog-help-links {
  flex-direction: column;
  justify-content: flex-start;
  gap: 2px;
}

.blog-help-links-wrapper {
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: 100%;
}

.blog-backgroundBorder {
  margin: 0;
  width: 100%;
  border-radius: 3px;
  background-color: #fff;
  border: 1px solid #efefef;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: flex-start;
  padding: 0 0 15px;
}

/*Blog Title*/

.blog-img-blog {
  height: 100%;
  width: 100%;
  position: absolute;
  margin: 0 !important;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: 3px;
  max-width: 100%;
  max-height: 100%;
  object-fit: cover;
}

.blog-gradient,
.blog-link {
  position: relative;
  max-width: 100%;
}

.blog-gradient {
  align-self: stretch;
  flex: 1;
  background: linear-gradient(180deg, transparent, rgba(51, 51, 51, 0.3) 90%);
  z-index: 1;
  cursor: pointer;
}

.blog-link {
  height: 192.8px;
  width: 342.8px;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-content: flex-start;
  min-width: 342.8px;
}

.blog-point,
.blog-views {
  position: relative;
  line-height: 20px;
  display: inline-block;
  min-width: 60px;
}

.blog-point {
  min-width: 5px;
}

.blog-views-parent {
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-content: flex-start;
  gap: 3.5px;
  font-size: 13px;
}

.blog-time-post {
  position: relative;
  line-height: 20px;
  display: inline-block;
  min-width: 83px;
  font-size: 12px;
}

.blog-frame-div {
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-content: flex-start;
  gap: 6.1px;
}

.blog-title {
  position: relative;
  font-size: 16px;
  line-height: 24px;
  font-weight: 600;
  color: #333;
  display: inline-block;
  min-width: 107.2px;
  cursor: pointer;
}

.blog-frame-container {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  gap: 10.4px;
  font-size: 13px;
}

.blog-content {
  margin: 0;
  font-size: 13.5px;
}

.blog-content-container,
.blog-read-more {
  position: relative;
  line-height: 26px;
}

.blog-read-more {
  line-height: 20px;
  font-weight: 550;
  color: #333;
  display: inline-block;
  min-width: 68.1px;
  font-size: 14px;
}

.blog-read-more:hover {
  color: #ed2a26;
}

.blog-arrow-icon {
  width: 13px;
  height: 13px;
  position: relative;
  flex-shrink: 0;
  margin-top: 2px;
}

.blog-arrow-icon-wrapper {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  padding: 2px 0 0;
}

.blog-read-more-parent {
  flex-direction: row;
  gap: 2.8px;
  color: #333;
  cursor: pointer;
}

.blog-read-more-parent:hover {
  color: #ed2a26;
}

.blog-backgroundBorder-1,
.blog-backgroundBorder-inner,
.blog-frame-group,
.blog-read-more-parent {
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
}

.blog-frame-group {
  flex-direction: column;
  gap: 14.8px;
  padding: 5px;
}

.blog-backgroundBorder-inner,
.blog-backgroundBorder-1 {
  box-sizing: border-box;
  max-width: 100%;
}

.blog-backgroundBorder-inner {
  flex-direction: column;
  padding: 4px 0 0;
  min-width: 470px;
}

.blog-backgroundBorder-1 {
  align-self: stretch;
  border-radius: 3px;
  background-color: #fff;
  border: 1px solid #efefef;
  flex-direction: row;
  padding: 9px 11px;
  gap: 20px;
  text-align: left;
  font-size: 14px;
  color: #686f7a;
}

.blog-backgroundBorder-parent,
.blog-backgroundBorder-group {
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  flex-shrink: 0;
}

.blog-backgroundBorder-group {
  flex: 1;
  flex-direction: column;
  gap: 30px;
  max-width: calc(100% - 300px);
}

.blog-backgroundBorder-parent {
  align-self: stretch;
  flex-direction: row;
  box-sizing: border-box;
  gap: 21px;
  max-width: 100%;
}

.blog-background-main {
  background: linear-gradient(#f7f7f7, #f7f7f7), #fff;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: flex-start;
  padding: 50px 150px 0;
}

.blog-prev {
  position: relative;
  line-height: 18px;
  display: inline-block;
  min-width: 35px;
}

.blog-link-previous {
  flex: 1;
  border-radius: 5.25px 3px 3px 5.25px;
  background-color: #fff;
  border: 1px solid #efefef;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-content: flex-start;
  padding: 6px 10px 7px 13px;
}

.blog-number-1,
.blog-link-1 {
  display: flex;
}

.blog-number-1 {
  width: 8.3px;
  position: relative;
  line-height: 17.5px;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
  min-width: 8.3px;
}

.blog-link-1 {
  width: 33.9px;
  border-radius: 3px;
  background-color: #333;
  border: 1px solid #333;
  box-sizing: border-box;
  flex-direction: row;
  align-items: flex-start;
  justify-content: flex-start;
  padding: 6px 11px 7px 13px;
  z-index: 1;
  margin-left: -1px;
  color: #fff;
}

.blog-number-2 {
  width: 8.3px;
  position: relative;
  line-height: 17.5px;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
  min-width: 8.3px;
}

.blog-link-2 {
  width: 33.9px;
  border-radius: 3px;
  background-color: #fff;
  border: 1px solid #efefef;
  box-sizing: border-box;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-content: flex-start;
  padding: 6px 11px 7px 13px;
  z-index: 2;
  margin-left: -1px;
}

.blog-number-3 {
  position: relative;
  line-height: 18px;
  display: inline-block;
  min-width: 12px;
}

.blog-link-3 {
  width: 37.1px;
  border-radius: 3px;
  background-color: #fff;
  border: 1px solid #efefef;
  box-sizing: border-box;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-content: flex-start;
  padding: 6px 10px 7px 13px;
  z-index: 3;
  margin-left: -1px;
}

.blog-number-4 {
  width: 16.1px;
  position: relative;
  line-height: 17.5px;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
  min-width: 16.1px;
}

.blog-link-4 {
  width: 41.7px;
  border-radius: 3px;
  background-color: #fff;
  border: 1px solid #efefef;
  box-sizing: border-box;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-content: flex-start;
  padding: 6px 11px 7px 13px;
  z-index: 4;
  margin-left: -1px;
}

.blog-next {
  position: relative;
  line-height: 18px;
  display: inline-block;
  min-width: 36px;
}

.blog-link-next,
.blog-nav-page-navigation-example {
  flex-direction: row;
  align-items: flex-start;
  justify-content: flex-start;
}

.blog-link-next {
  flex: 1;
  border-radius: 3px 5.25px 5.25px 3px;
  background-color: #fff;
  border: 1px solid #efefef;
  display: flex;
  padding: 6px 10px 7px 13px;
  z-index: 5;
  margin-left: -1px;
}

.blog-nav-page-navigation-example {
  border-radius: 3px;
  padding: 50px 0 20px 8px;
  flex-shrink: 0;
}

.blog-frame-parent,
.blog-nav-page-navigation-example,
.blog,
.blog-inner {
  display: flex;
  box-sizing: border-box;
}

.blog-frame-parent {
  flex-direction: column;
  align-items: flex-end;
  justify-content: flex-start;
  padding: 0;
  flex-shrink: 0;
  max-width: 100%;
  text-align: center;
  font-size: 14px;
  color: #686f7a !important;
}

.blog-inner {
  align-self: stretch;
  flex-direction: row;
  align-items: flex-start;
  justify-content: center;
  min-width: 100%;
  flex-shrink: 0;
}

.blog {
  width: 100%;
  position: relative;
  flex-direction: column;
  justify-content: flex-start;
  padding: 0;
  line-height: normal;
  letter-spacing: normal;
}

@media screen and (max-width: 1050px) {

   .blog-background-main {
    padding: 50px 100px 30px !important;
    
  }

  .blog-backgroundBorder-inner,
  .blog-link {
    flex: 1;
  }

  .blog-backgroundBorder-parent {
    display: block;
  }


  .blog-backgroundBorder-1 {
    flex-wrap: wrap;
  }

  .blog-backgroundBorder-group {
    max-width: 100%;
  }

  .blog-backgroundBorder{
    margin-bottom: 40px;
  }
}

@media screen and (max-width: 750px) {
  .blog-backgroundBorder-inner {
    min-width: 100%;
  }


  .blog-backgroundBorder-parent {
    padding-bottom: 213px;
    box-sizing: border-box;
  }

  .blog-frame-parent {
    gap: 26px;
  }

  .blog {
    gap: 25px;
  }
}
@media screen and (max-width: 450px) {
  .blog-navigation-links-parent {
    flex-wrap: wrap;
  }

  .blog-heading-2 {
    font-size: 19px;
    line-height: 24px;
  }

  .blog-heading-2-wrapper {
    padding-left: 20px;
    padding-right: 20px;
    box-sizing: border-box;
  }

  .blog-background {
    gap: 32px;
  }

  .blog-link {
    min-width: 100%;
  }

  .blog-help-links-wrapper {
    padding-left: 20px;
    padding-right: 20px;
    box-sizing: border-box;
  }

  .blog {
    height: auto;
  }
}
