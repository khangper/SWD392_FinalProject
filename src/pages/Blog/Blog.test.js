import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { renderWithProviders } from "../../constant/test-util";
import Blog from "../Blog/Blog";

describe("Blog", () => {
  beforeAll(() => {
    jest.spyOn(console, "error").mockImplementation(() => {});
    jest.spyOn(console, "log").mockImplementation(() => {});
  });

  afterAll(() => {
    console.error.mockRestore();
    console.log.mockRestore();
  });
  it("renders the Blog component", async () => {
    const { container } = renderWithProviders(<Blog />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});