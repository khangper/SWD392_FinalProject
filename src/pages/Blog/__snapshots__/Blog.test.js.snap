// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog renders the Blog component 1`] = `
<div>
  <div
    class="blog"
  >
    <header
      class="header"
    >
      <a
        href="/home"
      >
        <button
          class="header-link1"
        >
          Back To Cursus
        </button>
      </a>
      <a
        href="/home"
      >
        <img
          alt=""
          class="header-group-icon"
          src="https://gambolthemes.net/html-items/cursus-new-demo/images/logo.svg"
        />
      </a>
      <div
        class="profile"
      >
        <img
          alt=""
          class="header-avatar"
          src="profile-img.jpg"
        />
      </div>
    </header>
    <main
      class="blog-inner"
    >
      <div
        class="blog-frame-parent"
      >
        <div
          class="blog-background"
        >
          <div
            class="blog-navigation-links-parent"
          >
            <div
              class="blog-navigation-links"
            >
              <a
                class="blog-about"
                href="/about-us"
              >
                About
              </a>
            </div>
            <div
              class="blog-navigation-links-1"
            >
              <a
                class="blog-blog"
                href="/our-blog"
              >
                Blog
              </a>
            </div>
            <div
              class="blog-navigation-links-2"
            >
              <a
                class="blog-company"
                href="/company-details"
              >
                Company
              </a>
            </div>
            <div
              class="blog-navigation-links-3"
            >
              <a
                class="blog-careers"
                href="/career"
              >
                Careers
              </a>
            </div>
            <div
              class="blog-navigation-links-4"
            >
              <a
                class="blog-press"
                href="/press"
              >
                Press
              </a>
            </div>
          </div>
          <div
            class="blog-heading-2-wrapper"
          >
            <h2
              class="blog-heading-2"
            >
              Insights, ideas, and stories
            </h2>
          </div>
        </div>
        <div
          class="blog-background-main"
        >
          <div
            class="blog-backgroundBorder-parent"
          >
            <form
              class="blog-backgroundBorder"
            >
              <div
                class="blog-item"
              >
                <div
                  class="blog-input"
                >
                  <button
                    class="blog-button-search"
                  >
                    <svg
                      class="blog-icon-search"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 1024 1024"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                      />
                    </svg>
                  </button>
                  <input
                    class="blog-input-child"
                    placeholder="Search"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="blog-filter-item"
              >
                <div
                  class="blog-item-1"
                >
                  <div
                    class="blog-filter-dropdown"
                  >
                    <div
                      class="blog-filter-content"
                    >
                      <div
                        class="blog-labels"
                      >
                        Labels
                      </div>
                      <div
                        class="blog-symbol"
                      >
                        
                      </div>
                    </div>
                  </div>
                  <div
                    class="blog-filter-dropdown-1"
                  >
                    <div
                      class="blog-filter-content"
                    >
                      <div
                        class="blog-labels"
                      >
                        Archive
                      </div>
                      <div
                        class="blog-symbol"
                      >
                        
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="blog-item-2"
                >
                  <button
                    class="blog-button-1"
                  >
                    <svg
                      class="blog-icon-twitter"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                      />
                    </svg>
                    <div
                      class="blog-follow-twitter"
                    >
                      Follow
                    </div>
                  </button>
                  <button
                    class="blog-button-2"
                  >
                    <svg
                      class="blog-icon-facebook"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M15.4024 21V14.0344H17.7347L18.0838 11.3265H15.4024V9.59765C15.4024 8.81364 15.62 8.27934 16.7443 8.27934L18.1783 8.27867V5.85676C17.9302 5.82382 17.0791 5.75006 16.0888 5.75006C14.0213 5.75006 12.606 7.01198 12.606 9.32952V11.3265H10.2677V14.0344H12.606V21H4C3.44772 21 3 20.5523 3 20V4C3 3.44772 3.44772 3 4 3H20C20.5523 3 21 3.44772 21 4V20C21 20.5523 20.5523 21 20 21H15.4024Z"
                      />
                    </svg>
                    <div
                      class="blog-follow-facebook"
                    >
                      Follow
                    </div>
                  </button>
                </div>
              </div>
              <div
                class="blog-help-links-wrapper"
              >
                <div
                  class="blog-help-links"
                >
                  <div
                    class="blog-learn-more-link"
                  >
                    <p
                      class="blog-learn-more"
                    >
                      Learn More
                    </p>
                  </div>
                  <a
                    class="blog-help-center"
                    href="/"
                  >
                    Cursus Help Center
                  </a>
                </div>
              </div>
            </form>
            <div
              class="blog-backgroundBorder-group"
            />
          </div>
          <div
            class="blog-nav-page-navigation-example"
          >
            <div
              class="blog-link-previous"
            >
              <div
                class="blog-prev"
              >
                PREV
              </div>
            </div>
            <div
              class="blog-link-1"
            >
              <div
                class="blog-number-1"
              >
                1
              </div>
            </div>
            <div
              class="blog-link-2"
            >
              <div
                class="blog-number-2"
              >
                2
              </div>
            </div>
            <div
              class="blog-link-3"
            >
              <div
                class="blog-number-3"
              >
                ...
              </div>
            </div>
            <div
              class="blog-link-4"
            >
              <div
                class="blog-number-4"
              >
                24
              </div>
            </div>
            <div
              class="blog-link-next"
            >
              <div
                class="blog-next"
              >
                NEXT
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer
      class="footer footer-large-container"
    >
      <div
        class="footer-links"
      >
        <div
          class="footer-column"
        >
          <a
            class="link"
            href="/about-us"
          >
            About
          </a>
          <a
            class="link"
            href="/blog-single-view/:id"
          >
            Blog
          </a>
          <a
            class="link"
            href="/career"
          >
            Careers
          </a>
          <a
            class="link"
            href="/press"
          >
            Press
          </a>
        </div>
        <div
          class="footer-column"
        >
          <a
            class="link"
            href="/help"
          >
            Help
          </a>
          <a
            class="link"
            href="/coming-soon"
          >
            Advertise
          </a>
          <a
            class="link"
            href="/coming-soon"
          >
            Developers
          </a>
          <a
            class="link"
            href="/contactus"
          >
            Contact Us
          </a>
        </div>
        <div
          class="footer-column"
        >
          <a
            class="link"
            href="/termsofuse"
          >
            Copyright Policy
          </a>
          <a
            class="link"
            href="/termsofuse"
          >
            Terms
          </a>
          <a
            class="link"
            href="/termsofuse"
          >
            Privacy Policy
          </a>
          <a
            class="link"
            href="/sitemap"
          >
            Sitemap
          </a>
        </div>
        <div
          class="footer-column"
        >
          <button
            class="teach-button"
          >
            Teach on Cursus
          </button>
          <div
            class="language-selector"
          >
            <button>
              Language 
              <span>
                ▼
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="footer-bottom"
      >
        <div
          class="footer-copy"
        >
          <img
            alt=""
            src="https://gambolthemes.net/html-items/cursus-new-demo/images/logo1.svg"
          />
          © 2024 
          <strong>
            Cursus
          </strong>
          . All Rights Reserved.
        </div>
        <div
          class="footer-social"
        >
          <a
            href="/home"
          >
            <img
              alt=""
              class="social-icon"
              src="facebook.png"
            />
          </a>
          <a
            href="/home"
          >
            <img
              alt=""
              class="social-icon"
              src="x.png"
            />
          </a>
          <a
            href="/home"
          >
            <img
              alt=""
              class="social-icon"
              src="google+.png"
            />
          </a>
          <a
            href="/home"
          >
            <img
              alt=""
              class="social-icon"
              src="linkedin.png"
            />
          </a>
          <a
            href="/home"
          >
            <img
              alt=""
              class="social-icon"
              src="insta.png"
            />
          </a>
          <a
            href="/home"
          >
            <img
              alt=""
              class="social-icon"
              src="pinterest.png"
            />
          </a>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
